File: src/notifier.py
import requests
from .config import TELEGRAM_BOT_TOKEN, TELEGRAM_CHAT_ID

def send_telegram_message(message):
    """
    Sends a message to the configured Telegram chat.
    """
    if not TELEGRAM_BOT_TOKEN or not TELEGRAM_CHAT_ID:
        print("Telegram configuration missing. Skipping notification.")
        return False
        
    url = f"https://api.telegram.org/bot{TELEGRAM_BOT_TOKEN}/sendMessage"
    payload = {
        "chat_id": TELEGRAM_CHAT_ID,
        "text": message,
        "parse_mode": "Markdown"
    }
    
    try:
        response = requests.post(url, json=payload, timeout=10)
        response.raise_for_status()
        return True
    except Exception as e:
        print(f"Failed to send Telegram message: {e}")
        return False

File: src/notifier.ts
import { CONFIG } from "./config";

/**
 * Telegramにメッセージを送信します。
 * @param message 送信するメッセージ内容 (Markdown形式対応)
 * @param options 追加オプション (インラインキーボードなど)
 * @returns 送信成功時は true, 失敗時は false
 */
export async function sendTelegramMessage(
	message: string,
	options?: {
		reply_markup?: {
			inline_keyboard: { text: string; url: string }[][];
		};
	},
): Promise<boolean> {
	if (!CONFIG.TELEGRAM_BOT_TOKEN || !CONFIG.TELEGRAM_CHAT_ID) {
		console.error("Telegram configuration missing.");
		return false;
	}

	const url = `https://api.telegram.org/bot${CONFIG.TELEGRAM_BOT_TOKEN}/sendMessage`;
	const payload = {
		chat_id: CONFIG.TELEGRAM_CHAT_ID,
		text: message,
		parse_mode: "Markdown",
		...options,
	};

	try {
		const response = await fetch(url, {
			method: "POST",
			headers: {
				"Content-Type": "application/json",
			},
			body: JSON.stringify(payload),
		});

		if (!response.ok) {
			const errorText = await response.text();
			console.error(
				`Failed to send Telegram message: ${response.status} ${response.statusText}`,
				errorText,
			);
			return false;
		}

		return true;
	} catch (error) {
		console.error(`Failed to send Telegram message: ${error}`);
		return false;
	}
}
